### PR TITLE
Entity mutation in typescript

### PR DESCRIPTION
**Pull request by AliceLetitia**
Created a `mutate` method on entity handle. The `mutate` method transforms either a mutation callback function or new entity data into a series of entity operations and applies each operation to the model. 

The latest version of each individual field of the entity must now match the entity's latest version number. This means that whenever an operation is applied to an individual field, a `fastForward` operation must be applied to all other fields in the entity to update the version number.

### CODE REVIEW

**Comment by shans** [comment]
This looks really great! Everything seems basically correct; though I'll review in more detail on Friday, or you could ask Mick to take a look tomorrow?

File: src/runtime/storage/tests/reference-mode-store-test.ts [hunk 1]
```diff
@@ -172,9 +187,7 @@ describe('Reference Mode Store', async () => {
     const reference: Reference = {storageKey: new MockHierarchicalStorageKey(''), id: 'an-id', version: {[actor]: 1}};
     referenceCollection.applyOperation({type: CollectionOpTypes.Add, clock: {me: 1}, actor: 'me', added: reference});
 
-    const entityCRDT = new MyEntityModel();
-    entityCRDT.applyOperation({type: EntityOpTypes.Set, field: 'age', value: {id: '42', value: 42}, actor, clock: {[actor]: 1}});
-    entityCRDT.applyOperation({type: EntityOpTypes.Set, field: 'name', value: {id: 'bob'}, actor, clock: {[actor]: 1}});
+    const entityCRDT = myEntityToMyEntityModel(entity, actor);
```
Comment (shans) [inline src/runtime/storage/tests/reference-mode-store-test.ts:190]: you may not need this any more, now that fast-forwarding is built into the CRDT

**Review by shans — APPROVED** [review]

File: src/runtime/storage/tests/reference-mode-store-test.ts (hunk 1 above)
Comment (AliceLetitia) [reply src/runtime/storage/tests/reference-mode-store-test.ts:190]: Fast forwarding at the moment only updates the version number for all fields each time an operation is applied. When the fastForward operation can apply multiple operations at the same time, I'll replace the `meEntityToMyEntityModel` function with a fastForward operation. I plan to make this change in a follow up PR.

Comment (shans) [reply src/runtime/storage/tests/reference-mode-store-test.ts:190]: got it, thanks!

**Comment by AliceLetitia** [comment]
feedback/copybara failed check: I re-ran the one failing test and it passed in my CL.

**Pull request merged**